taskBuild.lua

```diff
@@ -485,3 +435,17 @@ function BuildEmpLauncer()
 	end
 	return unitName
 end
+
+--Function of function
+
+local function CommanderAA(tskqbhvr)
+	local unitName = DummyUnitName
+	if IsAANeeded() then
+		if ai.maphandler:IsUnderWater(tskqbhvr.unit:Internal():GetPosition()) then
```

Comment:
this is okay, but there are times a builder is on land and needs to build things on/under water, or is in water and needs to build things on land

one way to solve this would be to add both land and water building types to the queue seperately, and look for positions for each. another way would be to swap the unitnames later, as is done with mexes vs underwater mexes.

we don't need to solve it right away, but it has room for improvement